beluga_amcl/src/ndt_amcl_node_3d.cpp

```diff
@@ -0,0 +1,503 @@
+  }
+
+  {
+    auto descriptor = rcl_interfaces::msg::ParameterDescriptor();
+    descriptor.description = "Initial pose z axis covariance.";
+    this->declare_parameter("initial_pose.covariance_z", 1e-6, descriptor);
+  }
+
+  {
+    auto descriptor = rcl_interfaces::msg::ParameterDescriptor();
+    descriptor.description = "Initial pose roll axis covariance.";
+    this->declare_parameter("initial_pose.covariance_roll", 1e-6, descriptor);
+  }
+
+  {
+    auto descriptor = rcl_interfaces::msg::ParameterDescriptor();
+    descriptor.description = "Initial pose pitch axis covariance.";
+    this->declare_parameter("initial_pose.covariance_pitch", 1e-6, descriptor);
+  }
+
+  {
+    auto descriptor = rcl_interfaces::msg::ParameterDescriptor();
+    descriptor.description = "Likelihood for measurements that lie inside cells that are not present in the map.";
+    descriptor.floating_point_range.resize(1);
+    descriptor.floating_point_range[0].from_value = 0;
+    descriptor.floating_point_range[0].to_value = 1;
+    descriptor.floating_point_range[0].step = 0;
+    declare_parameter("minimum_likelihood", rclcpp::ParameterValue(0.01), descriptor);
+  }
+
+  {
+    auto descriptor = rcl_interfaces::msg::ParameterDescriptor();
+    descriptor.description = "Scaling parameter d1 in literature, used for scaling 3D likelihoods.";
+    descriptor.floating_point_range.resize(1);
+    descriptor.floating_point_range[0].from_value = 0;
+    descriptor.floating_point_range[0].to_value = 1000;
+    descriptor.floating_point_range[0].step = 0;
+    declare_parameter("d1", rclcpp::ParameterValue(1.0), descriptor);
+  }
+
+  {
+    auto descriptor = rcl_interfaces::msg::ParameterDescriptor();
+    descriptor.description = "Scaling parameter d2 in literature, used for scaling 3D likelihoods.";
+    descriptor.floating_point_range.resize(1);
+    descriptor.floating_point_range[0].from_value = 0;
+    descriptor.floating_point_range[0].to_value = 1000;
+    descriptor.floating_point_range[0].step = 0;
+    declare_parameter("d2", rclcpp::ParameterValue(0.6), descriptor);
+  }
+
+  if (get_parameter("autostart").as_bool()) {
+    auto autostart_delay = std::chrono::duration<double>(get_parameter("autostart_delay").as_double());
+    autostart_timer_ = create_wall_timer(autostart_delay, std::bind(&NdtAmclNode3D::autostart_callback, this));
+  }
+}
+
+void NdtAmclNode3D::do_autostart_callback() {
+  using lifecycle_msgs::msg::State;
+  auto current_state = configure();
+  if (current_state.id() != State::PRIMARY_STATE_INACTIVE) {
+    RCLCPP_WARN(get_logger(), "Failed to auto configure, shutting down");
+    shutdown();
+  }
+  RCLCPP_WARN(get_logger(), "Auto configured successfully");
+  current_state = activate();
+  if (current_state.id() != State::PRIMARY_STATE_ACTIVE) {
+    RCLCPP_WARN(get_logger(), "Failed to auto activate, shutting down");
+    shutdown();
+  }
+  RCLCPP_INFO(get_logger(), "Auto activated successfully");
+}
```

Comment:
@serraramiro1 meta: FWIW this looks like a perfectly reasonable autostart sequence for every node out there.